app/src/main/java/com/ncnf/organization/Organization.java

```diff
@@ -0,0 +1,178 @@
+package com.ncnf.organization;
+
+import com.google.firebase.firestore.GeoPoint;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+import com.ncnf.database.DatabaseResponse;
+import com.ncnf.database.DatabaseService;
+
+import static com.ncnf.Utils.ADDRESS_KEY;
+import static com.ncnf.Utils.ADMIN_KEY;
+import static com.ncnf.Utils.LOCATION_KEY;
+import static com.ncnf.Utils.NAME_KEY;
+import static com.ncnf.Utils.ORGANIZATIONS_COLLECTION_KEY;
+import static com.ncnf.Utils.ORGANIZED_EVENTS_KEY;
+import static com.ncnf.Utils.PHONE_NB_KEY;
+import static com.ncnf.Utils.UUID_KEY;
+import static com.ncnf.utilities.InputValidator.isArrayEmpty;
+import static com.ncnf.utilities.InputValidator.isStringEmpty;
+
+public class Organization {
+
+    private final UUID uuid;
+    private String name;
+    private GeoPoint location;
+    private String address;
+    private String phoneNumber;
+    private List<String> adminIds;
+    private List<String>  events;
+
+    public Organization(String name, GeoPoint location, String address, String phoneNumber, String originalOwner) {
+
+        this(UUID.randomUUID(), name, location, address, phoneNumber, new ArrayList<String>() {{ add(originalOwner); }}, new ArrayList<>());
+    }
+
+    public Organization(UUID uuid, String name, GeoPoint location, String address, String phoneNumber, List<String> adminIds, List<String> events) {
+        if(isArrayEmpty(adminIds) || !checkAdminIds(adminIds)){
```

Comment:
I guess those conditions are an illustration of the  "fail safe default", as I suppose those checks will be performed before adding an Organization to the DB. I suppose we will always add an Organization first to the DB with a single admin, namely the creator